static/app/utils/discover/eventView.tsx

```diff
@@ -535,6 +541,8 @@ class EventView {
         end: decodeScalar(end),
         statsPeriod: decodeScalar(statsPeriod),
         utc,
+        dataset:
+          (decodeScalar(location.query.dataset) as DiscoverDatasets) ?? saved.dataset,
```

Comment:
I checked to see if `dataset` was being serialized in the backend but I don't think I saw that it was. Since the description mentions "Open in Discover", we should also save this field somewhere. Is that tracked somewhere?

It looks like when we add some of these new filter type values, they go into a JSON blob on the model named `query`: https://github.com/getsentry/sentry/blob/master/src/sentry/discover/endpoints/serializers.py#L180

Reply:
Thanks for pointing that out @narsaynorath! Discussing this offline with the team to figure out what we want to do next with discover and new dataset. I'll merge this PR for now just to get things working in the meantime.